src/extension/content-script/batteries/SoundCloud.ts

```diff
@@ -4,16 +4,18 @@ import { findLightningAddressInText, setLightningData } from "./helpers";
 const urlMatcher =
   /^https:\/\/soundcloud.com\/([^/]+)(\/([^/]+))?(\/([^/]+))?(\/([^/]+))?$/;
 
+const pages = ["popular-tracks", "tracks", "albums", "sets", "reposts"];
 function battery(): void {
   const urlParts = document.location
     .toString()
     .replace("https://soundcloud.com", "")
     .split("/");
   const username = urlParts[1];
+  const page = urlParts[2];
 
-  if (username && !urlParts[2]) {
+  if (username && (!page || pages.includes(page))) {
     handleProfilePage();
-  } else if (urlParts[2] === "sets" || urlParts[3] === "sets") {
+  } else if (username && page) {
```

Comment:
why is this different now? sets are now done in `handleProfilePage`?

Reply:
It was a misunderstanding, https://soundcloud.com/monvol/sets is a different page that is similar to the other ones listed in the Array:
```TS
const pages = ["popular-tracks", "tracks", "albums", "sets", "reposts"];
```
 And if there's a `username/<something other than the ones in array>`, it is the track page